client/context.go

```diff
@@ -268,37 +276,37 @@ func (ctx Context) printOutput(out []byte) error {
 	return nil
 }
 
-// GetFromFields returns a from account address and Keybase name given either
+// GetFromFields returns a from account address, account name and keyring type, given either
 // an address or key name. If genOnly is true, only a valid Bech32 cosmos
 // address is returned.
-func GetFromFields(kr keyring.Keyring, from string, genOnly bool) (sdk.AccAddress, string, error) {
+func GetFromFields(kr keyring.Keyring, from string, genOnly bool) (sdk.AccAddress, string, keyring.KeyType, error) {
```

Comment:
probably would be better to return `(info, error)` here. All the return values are `info` attributes.

Reply:
My first reflex too, but not possible: "If genOnly is true, only a valid Bech32 cosmos address is returned.", that case doesn't return the keyring.